Apps/Database/src/Delegates/IResourceDelegateDelegate.cs

```diff
@@ -67,5 +68,33 @@ public interface IResourceDelegateDelegate
         /// <param name="delegateId">The delegated resource hdid.</param>
         /// <returns>A DB result which encapsulates the return record and status.</returns>
         bool Exists(string ownerId, string delegateId);
+
+        /// <summary>
+        /// Search resource delegates by criteria specified in the query.
+        /// </summary>
+        /// <param name="query">the query criteria.</param>
+        /// <returns>A <see cref="Task{TResult}"/> representing the result of the asynchronous operation.</returns>
+        public Task<ResourceDelegateQueryResult> Search(ResourceDelegateQuery query);
+    }
+
+    public record ResourceDelegateQuery
+    {
+        /// <summary>
+        /// Gets or sets search by resource delegate owner's hdid.
+        /// </summary>
+        public string? ByOwnerHdid { get; set; }
+
+        /// <summary>
+        /// Gets or sets search by resource delegate hdid.
+        /// </summary>
+        public string? ByDelegateHdid { get; set; }
+    }
+
+    public record ResourceDelegateQueryResult
+    {
+        /// <summary>
+        /// gets or sets the found items.
+        /// </summary>
+        public IEnumerable<ResourceDelegate> Items { get; set; } = Array.Empty<ResourceDelegate>();
     }
```

Comment:
We generally only define one type per file.  I'm not sure if adding these records is all that beneficial in general; maybe we can discuss the pattern with the team.